src/libraries/SqrtPriceMath.sol

```diff
@@ -47,10 +52,10 @@ library SqrtPriceMath {
             return uint160(UnsafeMath.divRoundingUp(numerator1, (numerator1 / sqrtPX96) + amount));
         } else {
             unchecked {
-                uint256 product;
+                uint256 product = amount * sqrtPX96;
                 // if the product overflows, we know the denominator underflows
                 // in addition, we must check that the denominator does not underflow
-                require((product = amount * sqrtPX96) / amount == sqrtPX96 && numerator1 > product);
+                if (product / amount != sqrtPX96 || numerator1 <= product) revert PriceOverflow();
```

Comment:
any concerns if amount is 0 for div?

Reply:
dont think so - thats in the [V3 core code too](https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/libraries/SqrtPriceMath.sol#L52)